File: src/main/java/core/exceptions/ArityException.java
package core.exceptions;

public class ArityException extends IllegalArgumentException {

  public ArityException(String name, int min, int max, int given) {
    super((name.isEmpty() ? "#<procedure>" : name) + ": arity mismatch; " +
           "the expected number of arguments does not match the given number " + "(" +
           "expected: " + ((min == max) ? min : ((max > 255) ? "at least " + min : min + " to " + max)) +
           ", given: " + given + ")", null);
  }

  @Override
  public synchronized Throwable fillInStackTrace() {
    return null;
  }
}

File: src/main/java/core/exceptions/ArityException.kt
package core.exceptions

import core.procedures.Arity

class ArityException(message: String) : IllegalArgumentException(message, null) {

    constructor(name: String, expected: Arity, given: Int) : this(name.ifEmpty { "#<procedure>" } +
            ": arity mismatch; the expected number of arguments does not match the given number (expected: $expected, given: $given)")

    @Synchronized override fun fillInStackTrace() = null
}
